lib/beaker/hypervisor/vagrant.rb

```diff
@@ -23,10 +23,13 @@ def private_network_generator(host)
       private_network_string = "    v.vm.network :private_network, ip: \"#{host['ip'].to_s}\", :netmask => \"#{host['netmask'] ||= "255.255.0.0"}\""
       case host['network_mac']
       when 'false'
+        @mac = randmac
         private_network_string << "\n"
       when nil
+        @mac = randmac
         private_network_string << ", :mac => \"#{randmac}\"\n"
```

Comment:
on line 32 below, I get a little twinge seeing the `host['network_mac']` used on two consecutive lines. This is because a) it's the tiniest bit wasteful to read the value from the host object again when you've just saved it in the `@mac` instance variable (not important) and b) if you want to change how you select the `@mac` value in the future, you need to make the change in two places rather than one. To be honest though, that's really an aesthetic objection that I wouldn't normally bring up or block merging on...

But in this case, wouldn't calling `randmac` twice end up with two separate values for the mac address?

Reply:
jep, good catch!